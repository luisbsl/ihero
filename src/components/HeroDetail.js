import React from 'react'
import { View, Text, Image } from 'react-native'
import styled from 'styled-components'

const Container = styled.View`
  flex: 1;
  flex-direction: column;
  align-items: center;
  width: 100%;
`

const TitleContainer = styled.View`
  height: 10%;
  width: 100%;
  justify-content: center;
  align-items: center;
`

const Title = styled.Text`
  font-size: 20px;
  font-weight: bold;
  color: #4d4d4d;
`

const ImageContainer = styled.View`
  height: 40%;
  width: 100%;
  justify-content: center;
  align-items: center;
`

const ImageStyled = styled.Image`
  width: 200px;
  height: 200px;
  border-radius: 100;
`

const DescriptionContainer = styled.View`
  height: 50%;
  width: 80%;
  justify-content: center;
  align-items: center;
`

const Description = styled.Text`
  color: #4d4d4d;
  font-size: 16px;
`

class HeroDetail extends React.Component {
  render() {
    const { hero } = this.props
    return (
      <Container>
        <TitleContainer>
          <Title>
            {hero.name}
          </Title>
        </TitleContainer>

        <ImageContainer>
          <ImageStyled
            resizeMode={'cover'}
            source={{ uri: `${hero.image}` }} />
        </ImageContainer>

        <DescriptionContainer>
          <Description>
            {hero.description? hero.description: 'No Description available'}
          </Description>
        </DescriptionContainer>
      </Container>
    )
  }
}

export default HeroDetail