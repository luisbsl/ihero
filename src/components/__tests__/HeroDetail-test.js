import React from 'react'
import 'jest-styled-components'
import renderer from 'react-test-renderer'

import HeroDetail from '../HeroDetail'

test('renders correctly', () => {
  const hero = {
    "id": "1009583",
    "image": "http://i.annihil.us/u/prod/marvel/i/mg/7/20/527bb5d64599e/standard_fantastic.jpg",
    "name": "She-Hulk (Jennifer Walters)",
    "description": "Seriously wounded, young lawyer Jennifer Walters was given a blood transfusion by her cousin Bruce Banner. Usually in better control of her powers and temper than the Hulk, She-Hulk has been a high profile New York lawyer as well as an upstanding member of both the Avengers and Fantastic Four."
  }

  const tree = renderer.create(<HeroDetail hero={hero} />).toJSON()
  expect(tree).toMatchSnapshot()
})