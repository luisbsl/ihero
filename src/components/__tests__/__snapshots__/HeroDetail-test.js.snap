// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexDirection": "column",
        "flexGrow": 1,
        "flexShrink": 1,
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "height": "10%",
          "justifyContent": "center",
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#4d4d4d",
            "fontSize": 20,
            "fontWeight": "bold",
          },
          undefined,
        ]
      }
    >
      She-Hulk (Jennifer Walters)
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "height": "40%",
          "justifyContent": "center",
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    <Image
      resizeMode="cover"
      source={
        Object {
          "uri": "http://i.annihil.us/u/prod/marvel/i/mg/7/20/527bb5d64599e/standard_fantastic.jpg",
        }
      }
      style={
        Array [
          Object {
            "borderRadius": 100,
            "height": 200,
            "width": 200,
          },
          undefined,
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "height": "50%",
          "justifyContent": "center",
          "width": "80%",
        },
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#4d4d4d",
            "fontSize": 16,
          },
          undefined,
        ]
      }
    >
      Seriously wounded, young lawyer Jennifer Walters was given a blood transfusion by her cousin Bruce Banner. Usually in better control of her powers and temper than the Hulk, She-Hulk has been a high profile New York lawyer as well as an upstanding member of both the Avengers and Fantastic Four.
    </Text>
  </View>
</View>
`;
